modules/common/condition/types.go

```diff
@@ -19,111 +19,65 @@ limitations under the License.
-	Message            string                 `json:"message,omitempty"`
-	LastHeartbeatTime  metav1.Time            `json:"lastHearbeatTime,omitempty"`
-	LastTransitionTime metav1.Time            `json:"lastTransitionTime,omitempty"`
-}
+const (
+	// SeverityError specifies that a condition with `Status=False` is an error.
+	SeverityError Severity = "Error"
+
+	// SeverityWarning specifies that a condition with `Status=False` is a warning.
+	SeverityWarning Severity = "Warning"
+
+	// SeverityInfo specifies that a condition with `Status=False` is informative.
+	SeverityInfo Severity = "Info"
+
+	// SeverityNone should apply only to conditions with `Status=True`.
+	SeverityNone Severity = ""
+)
 
 // Type - A summarizing name for a given condition
 type Type string
 
 // Reason - Why a particular condition is true, false or unknown
 type Reason string
 
-// NewCondition - Create a new condition object
-func NewCondition(
-	conditionType Type,
-	status corev1.ConditionStatus,
-	reason Reason,
-	message string,
-) Condition {
-	now := metav1.NewTime(time.Now().UTC().Truncate(time.Second))
-	condition := Condition{
-		Type:               conditionType,
-		Status:             status,
-		Reason:             reason,
-		Message:            message,
-		LastHeartbeatTime:  now,
-		LastTransitionTime: now,
-	}
-	return condition
-}
-
-// Set - Set a particular condition in a given condition list
-func (conditions *List) set(c *Condition) {
-	// Check for the existence of a particular condition type in a list of conditions
-	// and change it only if there is a status change
-	exists := false
-	for i := range *conditions {
-		existingCondition := (*conditions)[i]
-		if existingCondition.Type == c.Type {
-			exists = true
-			if !hasSameState(&existingCondition, c) {
-				(*conditions)[i] = *c
-				break
-			}
-			c.LastTransitionTime = existingCondition.LastTransitionTime
-			break
-		}
-	}
-
-	// If the condition does not exist, add it, setting the transition time only if not already set
-	if !exists {
-		*conditions = append(*conditions, *c)
-	}
-}
-
-// hasSameState returns true if a condition has the same state of another
-func hasSameState(i, j *Condition) bool {
-	return i.Type == j.Type &&
-		i.Status == j.Status &&
-		i.Reason == j.Reason &&
-		i.Message == j.Message
+// Condition defines an observation of a API resource operational state.
+type Condition struct {
+	// Type of condition in CamelCase.
+	Type Type `json:"type"`
+
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// Severity provides a classification of Reason code, so the current situation is immediately
+	// understandable and could act accordingly.
+	// It is meant for situations where Status=False and it should be indicated if it is just
+	// informational, warning (next reconciliation might fix it) or an error (e.g. DB create issue
+	// and no actions to automatically resolve the issue can/should be done).
+	// +optional
```

Comment:
Do we want to keep these fields optional? I can accept that Message being optional as the current Reason might not need further explanation, but I think Severity and Reason should always be provided. I also see that this patch provides a SeverityNone for the success case too.

Reply:
latest commit make sure TrueCondition and UnknownCondition set SeverityNone

Reply:
also removed optional tag

Reply:
SeverityNone works for me for Status=True.